src/rdkafka_admin.c

```diff
@@ -6666,3 +6773,705 @@ rd_kafka_DescribeConsumerGroups_result_groups(
+                                      size_t errstr_size) {
+        rd_kafka_metadata_internal_t *mdi = NULL;
+        struct rd_kafka_metadata *md      = NULL;
+        rd_kafka_resp_err_t err;
+        rd_list_t topics       = rko_req->rko_u.admin_request.args;
+        rd_kafka_broker_t *rkb = reply->rkbuf_rkb;
+        int i, cnt;
+        rd_kafka_op_t *rko_result = NULL;
+
+        err = rd_kafka_parse_Metadata(rkb, NULL, reply, &mdi, &topics);
+        if (err)
+                goto err;
+        rko_result = rd_kafka_admin_result_new(rko_req);
+        md         = &mdi->metadata;
+        rd_list_init(&rko_result->rko_u.admin_result.results, md->topic_cnt,
+                     rd_kafka_TopicDescription_free);
+        cnt = md->topic_cnt;
+        i   = 0;
+        while (cnt--) {
+                rd_kafka_TopicDescription_t *topicdesc = NULL;
+                if (md->topics[i].err == RD_KAFKA_RESP_ERR_NO_ERROR) {
+                        rd_list_t *authorized_operations;
+                        authorized_operations =
+                            rd_kafka_AuthorizedOperations_parse(
+                                mdi->topics[i].topic_authorized_operations);
+                        topicdesc = rd_kafka_TopicDescription_new(
+                            md->topics[i].topic, md->topics[i].partitions,
+                            md->topics[i].partition_cnt, authorized_operations,
+                            NULL);
+                        if (authorized_operations)
+                                rd_list_destroy(authorized_operations);
+                } else {
+                        rd_kafka_error_t *error =
+                            rd_kafka_error_new(md->topics[i].err, NULL);
+                        topicdesc = rd_kafka_TopicDescription_new_error(
+                            md->topics[i].topic, error);
+                        rd_kafka_error_destroy(error);
+                }
+                rd_list_add(&rko_result->rko_u.admin_result.results, topicdesc);
+        }
+        *rko_resultp = rko_result;
+        return RD_KAFKA_RESP_ERR_NO_ERROR;
+
+err:
+        if (rko_result)
+                rd_kafka_op_destroy(rko_result);
+        rd_snprintf(errstr, errstr_size,
+                    "DescribeTopics response protocol parse failure: %s",
+                    rd_kafka_err2str(reply->rkbuf_err));
+        return reply->rkbuf_err;
+}
+
+void rd_kafka_DescribeTopics(rd_kafka_t *rk,
+                             const char **topics,
+                             size_t topics_cnt,
+                             const rd_kafka_AdminOptions_t *options,
+                             rd_kafka_queue_t *rkqu) {
+        rd_kafka_op_t *rko_fanout;
+        rd_list_t dup_list;
+        size_t i;
+
+        static const struct rd_kafka_admin_fanout_worker_cbs fanout_cbs = {
+            rd_kafka_DescribeTopics_response_merge,
+            rd_kafka_TopicDescription_copy_opaque};
+
+        rd_assert(rkqu);
+
+        rko_fanout =
```

Comment:
Do we really need a fanout request here? I think we can live with request to controller for the metadata. Generally, all the brokers have metadata information present.